src/lib.rs

```diff
@@ -299,16 +299,19 @@ pub fn run(tool: Tool, matches: ArgMatches) -> Result<i32> {
         bail!("Unable to find workspace members");
     }
 
-    let target_artifact = if tool.needs_build() {
+    let mut tool_args = vec![];
+    if let Some(args) = matches.values_of("args") {
+        tool_args.extend(args);
+    }
+
+    let tool_help = tool_args.first() == Some(&"--help");
+
+    let target_artifact = if tool.needs_build() && !tool_help {
```

Comment:
would it make sense to check for `--help` in any position?

```suggestion
    let target_artifact = if tool.needs_build() && !tool_args.iter().any(|a| a == "--help") {
```

Reply:
I was thinking about it, but what if:

* `cargo size --help` - cargo help
* `cargo size -- --help` - size help
* `cargo size -- some args -- --help` - some deep inner help (like, for `ld`)

So, I'd stop with the original issue first.

Reply:
Alright, seems good!